ArcGISRuntimeSDKQt_CppSamples/Geometry/ClipGeometry/ClipGeometry.cpp

```diff
@@ -0,0 +1,128 @@
+
+  // Create an envelope for each clipping area
+  m_outsideEnvelope = Envelope(-12201990.219681, 5147942.225174, -11858344.321294, 5297071.577304, SpatialReference(3857));
+  m_intersectingEnvelope = Envelope(-12260345.183558, 4332053.378376, -11962086.479298, 4566553.881363, SpatialReference(3857));
+  m_insideEnvelope = Envelope(-11655182.595204, 4593570.068343, -11431488.567009, 4741618.772994, SpatialReference(3857));
+
+  // Set map to map view and append the two initial graphics overlays
+  m_mapView->setMap(m_map);
+  m_mapView->graphicsOverlays()->append(m_coloradoOverlay);
+  m_mapView->graphicsOverlays()->append(m_envelopesOverlay);
+
+  // Set initialViewpoint
+  Viewpoint initialViewpoint(Envelope(-12500000, 4000000, -11000000, 5500000, SpatialReference(3857)));
+  m_mapView->setViewpoint(initialViewpoint);
+
+  // Create the graphics
+  createGraphics();
+
+}
+
+// Creates the visual elements and adds them to their respective overlays
+void ClipGeometry::createGraphics(){
+
+  // Colorado outline + fill
+  m_coloradoOutline = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor("#FF0000FF"), 2.0f, this);
+  m_coloradoFill = new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor("#220000FF"), m_coloradoOutline, this);
+
+  // Create and add the Colorado graphic to its overlay
+  m_coloradoGraphic = new Graphic(m_coloradoEnvelope, m_coloradoFill, this);
+  m_coloradoOverlay->graphics()->append(m_coloradoGraphic);
+
+  // Clipping envelopes outline
+  SimpleLineSymbol* envelopeOutline = new SimpleLineSymbol(SimpleLineSymbolStyle::Dash, QColor("#FFFF0000"), 3.0f, this);
+
+  // Create and add the clipping envelope graphics to their overlay
+  m_outsideGraphic = new Graphic(m_outsideEnvelope, envelopeOutline, this);
+  m_intersectingGraphic = new Graphic(m_intersectingEnvelope, envelopeOutline, this);
+  m_insideGraphic = new Graphic(m_insideEnvelope, envelopeOutline, this);
+  m_envelopesOverlay->graphics()->append(m_outsideGraphic);
+  m_envelopesOverlay->graphics()->append(m_intersectingGraphic);
+  m_envelopesOverlay->graphics()->append(m_insideGraphic);
+
+}
+
+// Clips the overlapping geometries and displays them on the map
+void ClipGeometry::clipAreas(){
+
+  // Immediately hide the reference graphic
+  m_coloradoGraphic->setVisible(false);
+
+  // Get the graphicsListModel from the overlay containing the clipping envelopes
+  GraphicListModel* graphics = m_envelopesOverlay->graphics();
+
+  /* Iterate through the envelope graphics and call the static 'clip' function, passing in
+   * the reference graphic and the current iteration's envelope geometry as arguments.
+   * If the return value is valid, create a new graphic using the returned geometry and the
+   * fillSymbol of the reference graphic as arguments. Append the new graphic to its respective
+   * graphicsOverlay. */
+  for(int i = 0; i < graphics->size(); i++){
```

Comment:
update C++ styling to match our style used throughout samples. Also, you can cache graphics->size() to save that from needing to be called every time:

```cpp
const int graphicsSize = graphics->size();
for (int i = 0; i < graphicsSize; i++)
{
  ....
}
```